identity-iota/src/credential/errors.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::fmt::Display;
+
+use identity_core::common::OneOrMany;
+
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+/// An error associated with validating credentials and presentations.
+pub enum ValidationError {
+  /// Indicates that the expiration date of the credential is not considered valid.
+  #[error("the expiration date is in the past or earlier than required.")]
+  ExpirationDate,
+  /// Indicates that the issuance date of the credential is not considered valid.
+  #[error("the credential is yet to be active or has not been active for the required period.")]
+  IssuanceDate,
+  /// Indicates that the credential's signature could not be verified using the issuer's DID Document.
+  #[error("could not verify the issuer's signature")]
+  #[non_exhaustive]
+  IssuerProof {
+    source: Box<dyn std::error::Error + Send + Sync + 'static>, /* Todo: Would it be better to use a specific type
+                                                                 * * here? */
+  },
+  /// Indicates a failure to extract a resolved document corresponding to the URL of the credential's issuer upon
+  /// signature verification.
+  #[error("None of the supplied issuer documents correspond to the credential's issuer")]
+  #[non_exhaustive]
+  IncompatibleIssuerDocuments,
```

Comment:
Nit: not sure if we should lean into the trusted/untrusted terms.

```suggestion
  #[error("issuer did not match any of the supplied DID Documents")]
  #[non_exhaustive]
  UntrustedIssuer,
```